R/populateDefaults.R

```diff
@@ -0,0 +1,53 @@
+#' Generate a default settings shell based on settings metadata
+#'
+#' This function returns a default settings objectbased on the chart(s) specified. 
+#'
+#' The function is designed to work with valid safetyGraphics charts.
+#'
+#' @param charts The chart or chart(s) for which shells should be generated ("eDish" only for now) . Default: \code{"eDish"}.
+#' @return A list containing the appropriate settings for the selected chart
+#' 
+#' @examples 
+#' 
+#' generateShell(chart = "eDish") 
+#'  
+#' @keywords internal
+
+populateDefaults <- function(charts="eDish", settings ){ 
+  
+
+  chart="eDish" # expand too multiple charts
+  
+  defaultMappings <- safetyGraphics::getSettingsMetadata(
+    charts = charts, 
+    cols=c("text_key","default")
+  )  
+
+  hierarchical_metadata <- str_split(defaultMappings$text_key, "--") 
+  
+  
+  shell <- list()
+  #replace this !!!!!!!!!!!
+  for (i in 1:length(hierarchical_metadata) ) {
```

Comment:
this might even work without generating the shell - but this was my stuff from before - wasn't sure if it needed one of those cool recursive things you do sometimes haha but I suppose it's functional!

Reply:
I think I eliminated the call to this altogether ... Will double check and see if it makes sense.

Reply:
Confirmed that this function isn't needed.